services/rfq/relayer/limiter/limiter.go

```diff
@@ -0,0 +1,162 @@
+// Package limiter provides a rate limiting mechanism for RFQs to protect against reorgs.
+package limiter
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/synapsecns/sanguine/core/metrics"
+	"github.com/synapsecns/sanguine/ethergo/client"
+	"github.com/synapsecns/sanguine/services/rfq/relayer/quoter"
+	"github.com/synapsecns/sanguine/services/rfq/relayer/relconfig"
+	"github.com/synapsecns/sanguine/services/rfq/relayer/reldb"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+)
+
+// Limiter is the interface for rate limiting RFQs.
+type Limiter interface {
+	// IsAllowed returns true if the request is allowed, false otherwise.
+	IsAllowed(ctx context.Context, request reldb.QuoteRequest) (bool, error)
+}
+
+type limiterImpl struct {
+	client     client.EVM
+	metrics    metrics.Handler
+	quoter     quoter.Quoter
+	cfg        relconfig.Config
+	tokenNames map[string]relconfig.TokenConfig
+}
+
+// NewRateLimiter creates a new Limiter.
+// TODO: implement the sliding window: queue up requests and process them in order if cumulative volume is above limit.
+func NewRateLimiter(
+	cfg relconfig.Config,
+	q quoter.Quoter,
+	client client.EVM,
+	metricHandler metrics.Handler,
+	tokens map[string]relconfig.TokenConfig,
+) Limiter {
+	return &limiterImpl{
+		// Limiter:    ratelimit.New(cfg.MaxRFQSize),
+		client:     client,
+		metrics:    metricHandler,
+		quoter:     q,
+		cfg:        cfg,
+		tokenNames: tokens,
+	}
+}
+
+// IsAllowed returns true if the request is allowed, false otherwise.
+func (l *limiterImpl) IsAllowed(ctx context.Context, request reldb.QuoteRequest) (_ bool, err error) {
+	ctx, span := l.metrics.Tracer().Start(ctx, "limiter.IsAllowed", trace.WithAttributes(
```

Comment:
I think this is overkill, but that's more of a nit.

My bigger ask is if we're going to do this can we have a helper `quoteRequestToAttributes` func in some kinda utils file?

I have something similiar [here](https://github.com/synapsecns/sanguine/blob/085a75892aaeeaa2d30f164137e196855cc3254d/ethergo/submitter/util.go#L52-L87) for reference